lib/site.js

```diff
@@ -58,34 +63,43 @@ module.exports = class Site {
 			await mkdirp(this.target);
 		}
 
-		let { tree, layout } = this;
+		let { tree, renderers, generateLayout } = this;
 		if(!tree || filepaths === null) {
 			tree = await this.load();
 		}
 
+		let layout = generateLayout(tree);
+		let pageRenderer = new PageRenderer({ renderers, layout });
+
 		let dirs = new Set();
-		let writes = [];
-		tree.forEach(async page => {
-			let filepath = page.targetPath(this.target);
+		let writes = tree.map(async page => {
+			let filepath = path.resolve(this.target, page.slug);
```

Comment:
FWIW, I'd intentionally encapsulated this within `Page#targetDir` - so out of curiosity: Why did you extract it from there?

Reply:
It felt really akwards to have a method that takes a reference directory, just to resolve the path on that reference directory. So I decided to move this where the knowledge about the reference directory is.

Reply:
I was pondering the same thing, but came out on the other end, opting to disregard the implementation and focus on the semantics instead. But both works.